play/src/test/scala/net/yao/json/CollectionTest.scala

```diff
@@ -0,0 +1,150 @@
+package net.zhenglai
+package net.yao.json
+
+import org.scalatest.FunSuite
+
+import scala.collection.mutable
+
+class CollectionTest extends FunSuite {
+
+  test("immutable array") {
+    //    val ints: Array[Int] = new Array[Int](5)
+    val ints = new Array[Int](5)
+    assert(ints.length == 5)
+    assert(ints(0) == 0)
+
+    //use array companion object
+    // val doubles1 = Array[Double](1.0, 2.0)
+    val doubles = Array(1.0, 2.0)
+    assert(doubles(0) == 1.0)
+    doubles(0) = 3.0
+    assert(doubles(0) == 3.0)
+
+    for (i <- 0 to doubles.length - 1)
+      myPrint(doubles(i))
+    println()
+
+    for (i <- 0 until doubles.length)
+      myPrint(doubles(i))
+    println()
+
+    for (i <- doubles.indices)
+      myPrint(doubles(i))
+    println()
+
+    for (elem <- doubles)
+      myPrint(elem)
+    println()
+
+    val iterator = doubles.iterator
+    while (iterator.hasNext) {
+      myPrint(iterator.next())
+    }
+    println()
+
+    //    doubles.foreach((x: Double) => println(x))
+    doubles.foreach(myPrint)
+    println()
+
+    println(doubles.mkString(" === "))
+
+    //add
+    //    doubles :+ 4.0
+    val addTail = doubles.:+(4.0)
+    println(addTail.mkString(" === "))
+    assert(!(addTail sameElements doubles))
+
+    //    5.0 +: doubles
+    val addedHead = doubles.+:(5.0)
+    println(addedHead.mkString(" === "))
+
+  }
+
+  test("mutable array") {
+    //    val ints = new mutable.ArrayBuffer[Int]()
+    val ints = mutable.ArrayBuffer(1, 2, 3, 3)
+    println(ints)
+    println(ints.mkString("*"))
+
+    assert(ints(0) == 1)
+
+    ints += 12
+    println(ints)
+    val addedInts = ints.addOne(1)
+    println(ints)
+
+    assert(addedInts eq ints)
+    11 +=: ints
+    println(ints)
+
+    val array = ints.toArray
+    println(array.mkString("*"))
+
+    val buffer = array.toBuffer
+    println(buffer.mkString("-"))
+
+  }
+
+  test("dimensional array") {
+    val array = Array.ofDim[Int](2, 3)
+    println(array.map(_.mkString(" ")).mkString("\n"))
+
+    array.foreach(_.foreach(println))
```

Comment:
maybe you can implement a generic myPrint

Reply:
function `myPrint ` can be used in this case.

Reply:
make sense